x-pack/test/api_integration/apis/upgrade_assistant/upgrade_assistant.ts

```diff
@@ -121,5 +121,22 @@ export default function ({ getService }: FtrProviderContext) {
         expect(body.error).to.eql('Internal Server Error');
       });
     });
+
+    describe('GET /api/upgrade_assistant/status', () => {
+      it('returns a successful response', async () => {
+        const { body } = await supertest
+          .get('/api/upgrade_assistant/status')
+          .set('kbn-xsrf', 'xxx')
+          .expect(200);
+
+        const expectedResponseKeys = ['readyForUpgrade', 'details'];
+
+        // We're not able to easily test different upgrade status scenarios (there are tests with mocked data to handle this)
+        // so, for now, we simply verify the response returns the expected format
+        expectedResponseKeys.forEach((key) => {
+          expect(body[key]).to.not.equal(undefined);
```

Comment:
Nit; maybe we can test the kibana status at least? we have a `core_deprecations` functional plugin in `test/plugin_functional/plugins/core_plugin_deprecations` which we can use to test this endpoint in `test/plugin_functional/test_suites/upgrade_assistant`